app/routes/quotes/components/QuotePage.tsx

```diff
@@ -1,96 +1,110 @@
-import type Quote from 'app/store/models/Quote';
-import type { CurrentUser } from 'app/store/models/User';
-import type { ContentTarget } from 'app/store/utils/contentTarget';
+import { LoginPage } from 'app/components/LoginForm';
+import { selectEmojis } from 'app/reducers/emojis';
+import { selectQuoteById, selectQuotes } from 'app/reducers/quotes';
+import { selectPaginationNext } from 'app/reducers/selectors';
+import { useAppDispatch, useAppSelector } from 'app/store/hooks';
 import { navigation } from '../utils';
 import QuoteList from './QuoteList';
 import styles from './Quotes.css';
 
-type Props = {
-  query: {
-    approved: string;
-    ordering: string;
-  };
-  quotes: Quote[];
-  actionGrant: ActionGrant;
-  approve: (id: ID) => Promise<void>;
-  unapprove: (id: ID) => Promise<void>;
-  deleteQuote: (id: ID) => Promise<void>;
-  fetchAll: (args: {
-    query: {
-      approved: string;
-      ordering: string;
-    };
-    next?: boolean;
-  }) => Promise<void>;
-  showFetchMore: boolean;
-  currentUser: CurrentUser;
-  loggedIn: boolean;
-  addReaction: (args: {
-    emoji: string;
-    contentTarget: ContentTarget;
-  }) => Promise<void>;
-  deleteReaction: (args: {
-    reactionId: ID;
-    contentTarget: ContentTarget;
-  }) => Promise<void>;
-  fetching: boolean;
-  fetchEmojis: () => Promise<void>;
-  fetchingEmojis: boolean;
-  emojis: Emoji[];
-};
+const qsParamsParser = (search) => ({
+  approved:
+    qs.parse(search, {
+      ignoreQueryPrefix: true,
+    }).approved || 'true',
+  ordering: qs.parse(search, {
+    ignoreQueryPrefix: true,
+  }).ordering,
+});
+
 type Option = {
   label: string;
   value: string;
 };
+
 const filterRegDateOptions: Array<Option> = [
   {
     label: 'nyeste',
     value: '',
   },
   {
-    label: 'flest emojis',
+    label: 'flest reaksjoner',
     value: '?ordering=-reaction_count',
   },
 ];
-export default function QuotePage({
-  query,
-  quotes,
-  approve,
-  unapprove,
-  actionGrant,
-  deleteQuote,
-  fetchAll,
-  showFetchMore,
-  currentUser,
-  loggedIn,
-  addReaction,
-  deleteReaction,
-  emojis,
-  fetching,
-  fetchEmojis,
-  fetchingEmojis,
-}: Props) {
-  let errorMessage = undefined;
 
+type Props = {
+  loggedIn: boolean;
+};
+
+const QuotePage = ({ loggedIn }: Props) => {
+  const dispatch = useAppDispatch();
+
+  const { quoteId } = useParams();
+
+  const location = useLocation();
+
+  const { pagination } = useAppSelector((state) =>
+    selectPaginationNext({
+      endpoint: `/quotes/`,
+      query: qsParamsParser(location.search),
+      entity: 'quotes',
+    })(state)
+  );
+  const showFetchMore = pagination.hasMore;
+
+  const quotes = useAppSelector((state) => {
+    if (quoteId) {
+      return [selectQuoteById(state, quoteId)];
+    }
+
+    return selectQuotes(state, { pagination });
+  });
+
+  const fetching = useAppSelector((state) => state.quotes.fetching);
+
+  const query = qs.parse(location.search, {
+    ignoreQueryPrefix: true,
+  });
```

Comment:
The "load more" button doesn't work, because this qs-parsing doesn't add `approved=true`. Use the `qsParamsParser`-function:)

```suggestion
  const query = qsParamsParser(location.search);
```

Reply:
This also allows you to reuse the parsed qs on line 99

Reply:
Very nice catch!